addons/survey/wizard/survey_email_compose_message.py

```diff
@@ -78,44 +70,37 @@ def onchange_multi_email(self, cr, uid, ids, multi_email, context=None):
             raise UserError(_("One email at least is incorrect: %s" % error_message))
 
         emails_checked.sort()
-        values = {'multi_email': '\n'.join(emails_checked)}
-        return {'value': values}
+        self.multi_email = '\n'.join(emails_checked)
 
-    def onchange_survey_id(self, cr, uid, ids, survey_id, context=None):
+    @api.onchange('survey_id')
+    def onchange_survey_id(self):
         """ Compute if the message is unread by the current user. """
-        if survey_id:
-            survey = self.pool.get('survey.survey').browse(cr, uid, survey_id, context=context)
-            return {
-                'value': {
-                    'subject': survey.title,
-                    'public_url': survey.public_url,
-                    'public_url_html': '<a href="%s">%s</a>' % (survey.public_url, _("Click here to take survey")),
-                }}
+        if self.survey_id:
+            self.subject = self.survey_id.title
+            self.public_url = self.survey_id.public_url
+            self.public_url_html = '<a href="%s">%s</a>' % (self.survey_id.public_url, _("Click here to take survey"))
         else:
             txt = _("Please select a survey")
-            return {
-                'value': {
-                    'public_url': txt,
-                    'public_url_html': txt,
-                }}
+            self.public_url = txt
+            self.public_url_html = txt
 
     #------------------------------------------------------
     # Wizard validation and send
     #------------------------------------------------------
 
-    def send_mail(self, cr, uid, ids, auto_commit=False, context=None):
+    @api.multi
+    def send_mail(self, auto_commit=False):
         """ Process the wizard content and proceed with sending the related
             email(s), rendering any template patterns on the fly if needed """
-        if context is None:
-            context = {}
 
-        survey_response_obj = self.pool.get('survey.user_input')
-        partner_obj = self.pool.get('res.partner')
-        mail_mail_obj = self.pool.get('mail.mail')
+        self.ensure_one()
+        UserInput = self.env['survey.user_input']
+        Partner = self.env['res.partner']
+        Mail = self.env['mail.mail']
         try:
-            model, anonymous_id = self.pool.get('ir.model.data').get_object_reference(cr, uid, 'portal', 'group_anonymous')
+            anonymous = self.env.ref('portal.group_anonymous')
```

Comment:
``` anonymous = self.env.ref('portal.group_anonymous', raise_if_not_found=False) ```
Should do the job, if group_anonymous is not found then it will return None, so no need of try..except